src/odemis/acq/stitching/_weaver.py

```diff
@@ -57,12 +56,45 @@ def addTile(self, tile):
         img.mergeMetadata(tile.metadata)
         self.tiles.append(tile)
 
+    @abstractmethod
     def getFullImage(self):
         """
-        return (2D DataArray): same dtype as the tiles, with shape corresponding to the bounding box. 
+        Assembles the tiles into a large image.
+        return (2D DataArray): same dtype as the tiles, with shape corresponding to the bounding box.
         """
+        pass
 
-        tiles = self.tiles
+    def _adjust_brightness(self, tiles):
+        """
+        Adjusts the brightness of a list of tiles, so they all have the same mean value.
+        tiles (2D DataArray): input tiles
+        return (2D DataArray): tiles with adjusted brightness
+        """
+        im_brt = numpy.mean(tiles)
```

Comment:
Please add a comment that this is not the best algorithm ever (because in reality, not every tile has the same average intensity). And one possibility would be to have local brightness adjustments, for instance by comparing the overlapping region (but with bleaching, it's not so easy).